src/components/Switch/Switch.tsx

```diff
@@ -69,37 +67,65 @@ export const Switch: FC<SwitchProps> = ({
     const trackClasses = useMemo(() => {
         const baseClasses = 'tw-group tw-border tw-inline-flex tw-rounded-full tw-shrink-0 tw-p-0 tw-transition-colors';
 
-        const valueClasses = on
-            ? 'tw-bg-text-weak tw-border-line-xx-strong hover:tw-bg-text'
-            : 'tw-bg-box-neutral tw-border-line-x-strong hover:tw-bg-box-neutral-hover';
+        const trueOrFalseValueClasses =
+            mode === 'on'
+                ? 'tw-bg-text-weak tw-border-line-xx-strong hover:tw-bg-text'
+                : 'tw-bg-box-neutral tw-border-line-x-strong hover:tw-bg-box-neutral-hover';
+
+        const valueClasses =
+            mode === 'indeterminate'
+                ? 'tw-bg-text-weak tw-flex tw-items-center tw-justify-center hover:tw-bg-text'
+                : trueOrFalseValueClasses;
+
+        const trueOrFalseDisabledClasses = disabled
+            ? 'tw-bg-box-disabled tw-border-line tw-pointer-events-none'
+            : valueClasses;
 
-        const disabledClasses = disabled ? 'tw-bg-box-disabled tw-border-line tw-pointer-events-none' : valueClasses;
+        const disabledClasses =
+            disabled && mode === 'indeterminate'
+                ? 'tw-flex tw-items-center tw-justify-center tw-bg-box-disabled tw-border-line tw-pointer-events-none'
+                : trueOrFalseDisabledClasses;
 
         return merge([baseClasses, disabledClasses, trackSizeClasses[size], isFocusVisible && FOCUS_STYLE]);
-    }, [on, disabled, size, isFocusVisible]);
+    }, [mode, disabled, size, isFocusVisible]);
 
     // Responsible for the left-right translation
     const dotWrapperClasses = useMemo(() => {
         const baseClasses = 'tw-relative tw-self-center tw-transition-transform';
 
-        const valueClasses = on && dotSizeClasses[size].activeTranslation;
+        const trueOrFalseValueClasses = mode === 'on' ? dotSizeClasses[size].activeTranslation : '';
 
-        return merge([baseClasses, dotSizeClasses[size].dimensions, valueClasses]);
-    }, [on, size]);
+        const valueClasses = mode === 'indeterminate' ? 'tw-bg-base' : trueOrFalseValueClasses;
+
+        const disabledClasses = disabled && mode === 'indeterminate' ? 'tw-bg-text-disabled' : valueClasses;
+
+        return merge([
+            baseClasses,
+            mode === 'indeterminate' ? indeterminateLineClasses : dotSizeClasses[size].dimensions,
+            disabledClasses,
+        ]);
+    }, [mode, size]);
```

Comment:
`React Hook useMemo has a missing dependency: 'disabled'. Either include it or remove the dependency array`